### PR TITLE
[ZEPPELIN-3837] update how_to_build.md

### PR DESCRIPTION
**Pull request by Leemoonsoo**
### What is this PR for?
how_to_build.md includes some build profiles that has removed/changed.
This PR makes how_to_build.md up to dated.

### What type of PR is it?
Documentation

### What is the Jira issue?
https://issues.apache.org/jira/browse/ZEPPELIN-3837

### Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? no
* Does this needs documentation? no

### CODE REVIEW

File: docs/setup/basics/how_to_build.md [hunk 1]
```diff
@@ -70,7 +70,7 @@ If you're unsure about the options, use the same commands that creates official
 # update all pom.xml to use scala 2.11
 ./dev/change_scala_version.sh 2.11
 # build zeppelin with all interpreters and include latest version of Apache spark support for local mode.
-mvn clean package -DskipTests -Pspark-2.0 -Phadoop-2.4 -Pr -Pscala-2.11
+mvn clean package -DskipTests -Pspark-2.1 -Phadoop-2.6 -Pscala-2.11
```
[the remark this answers is not in the document]
Comment (Leemoonsoo) [reply docs/setup/basics/how_to_build.md]: I was trying to match it with https://github.com/apache/zeppelin/blob/master/dev/create_release.sh#L109. Would it be better to change 2.3 on both places?

[the remark this answers is not in the document]
Comment (felixcheung) [reply docs/setup/basics/how_to_build.md]: possibly..

[the remark this answers is not in the document]
Comment (Leemoonsoo) [reply docs/setup/basics/how_to_build.md]: Pushed a commit to update `-Pspark-2.1` -> `-Pspark-2.3` on both `dev/create_release.sh` and `docs/setup/basics/how_to_build.md`.

**Comment by Leemoonsoo** [comment]
Merge to master

**Pull request closed without merging**